w07-ruby-on-rails-cont/d3-homework/README.md

```diff
@@ -0,0 +1,88 @@
+
+1. You clone the project John pushed up to Github after he makes you a collaborator. This creates a local repository with a `remote` called `origin` which is the original repo John created and that is technically owned by John. From within your local repository, you can check the urls of all remote repos it's connected to.
+  ```
+  $ git remote -v
+  ```
+
+2. Make a new branch called `auth`. Feature branches help organize work on a project.
+  ```
+    $ git checkout -b auth
+  ```
+
+3. Now you build the entire auth feature with sign up, login, logout, etc. How do you get your new code up to the main group project?
+
+4. Switch back to the local/master branch, then pull from origin/master to bring your local master up to date.
+  ```
+    $ git checkout master
+    $ git pull origin master
+  ```
+
+5. Now switch back to your local/auth branch and merge local/master into it. This will give you a chance to resolve all conflicts with the most up to date master version before pushing your branch to origin/master. Watch git's output in your terminal to see which files have merge conflicts.
+  ```
+    $ git checkout auth
+    $ git merge master
+  ```
+
+6. Once you have resolved all the conflicts, add and commit your changes locally and push your feature branch to the remote repo.
+  ```
+    $ git push origin auth
+  ```
+
+7. Now navigate in your browser to original remote repo's GitHub page. GitHub will detect that you just pushed a new branch and give you a prompt to make that branch into a pull request. Go for it! This will notify all the collaborators that a pull request is waiting to be merged. In industry, the next step would be for someone else on the team to review your code and suggest any changes needed before it's merged in.  As soon as you've created the pull request, though, John or another member of the team can try to merge it in.
+
+8. If there are no other pull requests in the queue, John can merge your pull request in cleanly with the most up to date master. Otherwise, John might choose to merge a few pull requests before yours, and those pull requests might create merge conflicts with yours. In that case, GitHub will show that there are merge conflicts. John can pull down your pull request branch and merge them locally to remove all conflicts, or he can ask you to and you can repeat the last few steps here (starting with `git pull origin master` in your local repo) to get the newest changes from the group repo and then merge again. Remember its not the owner of git success to fix all merge conflicts! (How could they do that? They didn't write the code! Each branch owner is responsible to get a clean pull request queued up)
```

Comment:
Remember its not the owner of git success to fix all merge conflicts!
Remember its not the owner of the git repo's responsibility to resolve all merge conflicts!